### DISCUSSION TITLE
Anything to uncompressed stream

### DISCUSSION THREAD

**Discussion by DimHDcode** [CESNET/UltraGrid#185] [Q&A]
I'm trying to create an uncompressed stream (for example, with testgrid) in order to open it on the same computer:
uv -t testgrid 10.0.0.98

10.0.0.98 - Microsoft LoopBack adapter

Then I try to open it in VLC:
udp://10.0.0.98:5004

VLC thinks for a long time, but nothing happens. What am I doing wrong?
Thanks.

**Comment by MartinPulec** [comment]
This is not possible to do it like this now. There are basically these points:

1. signalization - uncompressed video ([RFC 4175](https://datatracker.ietf.org/doc/html/rfc4175)) AFAIK requires explicit signalization (usually SDP), which would require use of eg. RTSP (or providing SDP file otherwise)
2. packet format - UltraGrid supports only RFC-compliant packet formats for JPEG and H.264 (using `--video-protocol RTSP|SDP` is required)

What do you want to achieve? The point _1_ can be avoided by using JPEG (JPEG doesn't need to be signalized, it has registered RTP packet type), eg.:

    uv -t testcard --video-protocol SDP -c libavcodec:subsampling=420 127.0.0.1
    ffplay rtp://127.0.0.1:5004

Anyways, I didn't manage to persuade VLC to consume that stream. In theory, RFC-compliant uncompressed video could be implemented but I doubt it would be useful.
